pkg/tbtc/node.go

```diff
@@ -65,6 +66,24 @@ type node struct {
 	// dkgExecutor MUST NOT be used outside this struct.
 	dkgExecutor *dkgExecutor
 
+	heartbeatFailureCountersMutex sync.Mutex
+	// heartbeatFailureCounters holds counters keeping track of consecutive
+	// heartbeat failures. Each wallet has a separate counter. The key used in
+	// the map is the uncompressed public key (with 04 prefix) of the wallet.
+	heartbeatFailureCounters map[string]*uint
```

Comment:
What do you think about introducing a specialized structure `heartbeatFailureCounter` and passing it to heartbeat actions instead of bare `*uint`? The structure would look as follows:

```
type heartbeatFailureCounter struct {
 mutex sync.Mutex
 counters map[string]uint // uncompressed public key (with 04 prefix) of the wallet as key
}

func (hfc *heartbeatFailureCounter) increment(walletPublicKey string)
func (hfc *heartbeatFailureCounter) reset(walletPublicKey string)
func (hfc *heartbeatFailureCounter) get(walletPublicKey string)
```

The main reason is that dealing with bare `*uint` is error-prone. We always need to dereference the pointer in order to update the underlying value. It's not so hard to make a runtime-level mistake here while doing some changes in the future. Second reason is extensibility. If we ever need to persist the counter to survive client restarts, we have everything in one place and the task will be much easier.

WDYT?

Reply:
Done in 2695f771739aa688487a89ce29122765342709de.
I wanted to keep it simple and didn't validate the string argument in each function.
We could in theory validate the length of the string argument, but it would make using the counter unwieldy - we would have to check for error every time we use the counter.
Perhaps it would be better to have a structure `heartbeatFailureCounter` for each wallet:
```
type heartbeatFailureCounter struct {
   counter int
}

func (hfc *heartbeatFailureCounter) increment()
func (hfc *heartbeatFailureCounter) reset()
func (hfc *heartbeatFailureCounter) get()
```
and store the map of these objects in a map in `node`?
WDYT?